### DISCUSSION TITLE
dependentRequires

### DISCUSSION THREAD

**Discussion by k-channon-PA** [pboettch/json-schema-validator#281] [Q&A]
Hi, is the "dependentRequired" keyword implemented?  I have an example that works in the online validator at www.jsonschemavalidator.net, but I can't seem to get it to work in C++ using this library (which I'm successfully using for a bunch of other things):
```json
{
  "type": "object",
  "properties": {
    "foo": { "type": "string" },
    "bar": { "type": "string" }
  },
  "dependentRequired": {
    "bar": ["foo"]
  }
}
```
So, this should allow `{"foo":"hello!"}` and `{"foo": "hello!", "bar": "howdy!"}`, but not a thing with `bar` on its own, like: `{"bar": "howdy!"}`

Am I just holding it wrong?
using v2.1.0

Here's an example (using Catch2)

```c++
TEST_CASE("bar is not allowed without foo")
{
  using nlohmann::json;

  nlohmann::json_schema::json_validator validator(nullptr, nlohmann::json_schema::default_string_format_check);
  validator.set_root_schema(json::parse(R"({
    "type": "object",
    "properties": {
      "foo": { "type": "string" },
      "bar": { "type": "string" }
    },
    "dependentRequired": {
      "bar": ["foo"]
    }
    })")
    );

  REQUIRE_NOTHROW(validator.validate(json::parse(R"({"foo":"hello!"})")));
  REQUIRE_NOTHROW(validator.validate(json::parse(R"({"foo":"hello!", "bar":"howdy!"})")));
  
  // FAILS :(
  REQUIRE_THROWS(validator.validate(json::parse(R"({"bar":"howdy!"})")));
}
```

**Comment by pboettch** [comment]
Not known by draft7, which this library supports, so not implemented. Sorry.
